### PR TITLE
Fix behaviour of unfrozen BatchNormalization layer (resolves #46)

### LINKED ISSUE #46: Training ResNet50 on ImageNet does not achieve reasonable performance
I previously had no problems training the ResNet50 implementation bundled with Keras in `keras.applications.resnet50` to 70% validation accuracy on the ILSVRC 2012 dataset.

Now I wanted to switch to the `keras_resnet` implementation, but was not able to get validation accuracy above 30%. Right after the first epoch, accuracy of `keras_resnet` is about 1%, while the bundled `ResNet50` already achieves 11%.

I am creating the ResNet like this:

    input_ = keras.layers.Input((3, None, None)) if K.image_data_format() == 'channels_first' else 
    keras.layers.Input((None, None, 3))
    rn = keras_resnet.models.ResNet50(input_, include_top = True, classes = 1000, freeze_bn = False)

I've already tried different learning rate schedules and optimizers, but nothing worked.

Is there anything special I have to take care of?

### PR DESCRIPTION
**Pull request by Callidior**
Previously, if `BatchNormalization` was initialized with `BatchNormalization(freeze=False)`, its behaviour was not equivalent to the standard `BatchNormalization` layer, as one would expect. Instead, it was always forced to be in training mode, providing wrong validation results.

This PR does not change the behaviour for `freeze=True`, but makes the layer equivalent to the standard `BatchNormalization` layer from Keras for `freeze=False`.

### CODE REVIEW

**Comment by hgaiser** [comment]
Doesn't this *only* change the behaviour if `freeze=True`?

Also, what accuracy are you getting now?

**Comment by Callidior** [comment]
No, the behaviour for `freeze=True` is not changed. Previously, we called the method of the superclass with `training=(not self.freeze)`, which would evaluate to `training=False`. Now, if `self.freeze is True`, we set `training=False`, as before.

If `self.freeze is False`, however, we now have `training=None` (the default) instead of `training=True`.

It's still training, but I now already have 12% validation accuracy after the first epoch and 40% after 4 epochs, which is already higher than anything I got without the modifications made in this PR.

**Comment by Callidior** [comment]
By the way, I would question the example in the README. The model is initialized there with `freeze_bn=True`, which fixes the `BatchNormalization` layers to test using the initialization parameters. This should be equivalent to using no batch normalization at all.

I also tried this first for my ImageNet training, since the README does so, but it didn't work.

**Comment by Callidior** [comment]
I now finally obtained 68% validation accuracy, which is much closer to what I got with the bundled ResNet-50 than before.

**Comment by 0x00b1** [comment]
Awesome. Thanks, @Callidior.
